comment/tests/test_mixins.py

```diff
@@ -16,131 +21,179 @@ def test_non_ajax_request(self):
+        self.assertEqual(response.status_code, 403)
+
+    def test_delete_flagged_comment_by_moderator(self):
+        """moderator cannot delete comment unless it's flagged"""
+        self.assertNotEqual(self.comment.user.id, self.moderator.id)
+        self.client.force_login(self.moderator)
+        self.create_flag_instance(self.user_2, self.comment)
+        self.create_flag_instance(self.admin, self.comment)
+        self.create_flag_instance(self.moderator, self.comment)
+        self.assertEqual(self.flags, 3)
+        self.assertTrue(self.comment.is_flagged)
+        response = self.client.post(self.url, data=self.data, HTTP_X_REQUESTED_WITH='XMLHttpRequest')
+        self.assertEqual(response.status_code, 200)
+
+
+class BaseFlagMixinTest(BaseCommentMixinTest):
+    def setUp(self):
+        super().setUp()
+        self.client.force_login(self.user_1)
+        self.comment = self.create_comment(self.post_1, self.user_1)
+        self.url = reverse('comment:flag', kwargs={'pk': self.comment.id})
 
+    @patch('comment.mixins.settings', COMMENT_FLAGS_ALLOWED=0)
+    def test_flag_not_enabled_permission(self, _):
+        """permission denied when flagging not enabled"""
+        self.client.force_login(self.user_2)
+        response = self.client.post(self.url, HTTP_X_REQUESTED_WITH='XMLHttpRequest')
+        self.assertEqual(response.status_code, 403)
 
-class TestParentIdMixin(BaseCommentMixinTest):
-    @classmethod
-    def setUpTestData(cls):
-        super().setUpTestData()
-        cls.create_comment(cls.content_object_1)
-        cls.create_comment(cls.content_object_2)
-        cls.mixin = ParentIdMixin()
-        cls.mixin.api = True
-        cls.base_url = '/api/comments/create/'
-
-    def test_parent_id_not_int(self):
-        # parent id not int
-        parent_id = 'c'
-        url_data = self.url_data.copy()
-        url_data['parent_id'] = parent_id
-        request = self.factory.get(self.get_url(**url_data))
-        response = self.mixin.dispatch(request)
+    @patch('comment.mixins.settings', COMMENT_FLAGS_ALLOWED=2)
+    def test_flag_enabled_permission(self, _):
+        self.client.force_login(self.user_2)
+        response = self.client.post(self.url, HTTP_X_REQUESTED_WITH='XMLHttpRequest')
+        self.assertEqual(response.status_code, 200)
 
-        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
-        self.assertEqual(self.mixin.error, f'the parent id must be an integer, {parent_id} is NOT')
 
-    def test_parent_id_does_not_exist(self):
-        parent_id = 100
-        url_data = self.url_data.copy()
-        url_data['parent_id'] = parent_id
-        request = self.factory.get(self.get_url(**url_data))
-        response = self.mixin.dispatch(request)
+class CanSetFlagMixin(BaseCommentMixinTest):
+    def setUp(self):
+        super().setUp()
+        self.client.force_login(self.user_1)
+        self.comment = self.create_comment(self.post_1, self.user_1)
```

Comment:
Even `create_comment` is a `classmethod` now !

Also consider moving this to `setUpTestData`